### PR TITLE
Support for Vert.x RxJava

### PR DESCRIPTION
**Pull request by khadalasfx**
Added instrumentation for AsyncResultSingle. This fixes the issue with reactive handlers causality.

### CODE REVIEW

File: dd-java-agent/instrumentation/vertx/src/main/java/datadog/trace/instrumentation/vertx/VertxRxInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+      packageName + ".AsyncResultSingleAdvice", packageName + ".AsyncResultHandlerConsumerWrapper"
+    };
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    final Map adviceMap = new HashMap();
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/vertx/src/main/java/datadog/trace/instrumentation/vertx/VertxRxInstrumentation.java:41]: Nit: the pattern has been to use `singletonMap`

File: dd-java-agent/instrumentation/vertx/src/rxHandlerTests/groovy/VertxRxServerTest.groovy [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import io.vertx.reactivex.core.Vertx
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import spock.lang.Shared
+
+import static datadog.trace.agent.test.base.HttpServerTest.ServerEndpoint.SUCCESS
+
+class VertxRxServerTest extends AgentTestRunner {
+  static {
+    System.setProperty("dd.integration.jdbc.enabled", "true")
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/vertx/src/rxHandlerTests/groovy/VertxRxServerTest.groovy:16]: Not jumping out to me why this is necessary if the instrumentation is included with the test task.

File: dd-java-agent/instrumentation/vertx/vertx.gradle [hunk 3]
```diff
@@ -56,11 +60,19 @@ dependencies {
   handlerTestCompile group: 'io.vertx', name: 'vertx-core', version: '3.1.0'
   handlerTestCompile group: 'io.vertx', name: 'vertx-web', version: '3.0.0'
 
+  rxHandlerTestCompile project(':dd-java-agent:instrumentation:netty-4.0')
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/vertx/vertx.gradle:63]: doesn't vertx-core 3.6 use netty 4.1?

File: dd-java-agent/instrumentation/vertx/src/main/java/datadog/trace/instrumentation/vertx/VertxRxInstrumentation.java (hunk 1 above)
Comment (khadalasfx) [reply dd-java-agent/instrumentation/vertx/src/main/java/datadog/trace/instrumentation/vertx/VertxRxInstrumentation.java:41]: Fixed

File: dd-java-agent/instrumentation/vertx/src/rxHandlerTests/groovy/VertxRxServerTest.groovy (hunk 2 above)
Comment (khadalasfx) [reply dd-java-agent/instrumentation/vertx/src/rxHandlerTests/groovy/VertxRxServerTest.groovy:16]: I don't know all details how this flag is used, but it is even necessary in https://github.com/signalfx/signalfx-java-tracing/blob/3f8a80189bf942b9ac32c419cd0741cd8ef5e8cf/dd-java-agent/instrumentation/jdbc/src/test/groovy/JDBCInstrumentationTest.groovy#L30

File: dd-java-agent/instrumentation/vertx/vertx.gradle (hunk 3 above)
Comment (khadalasfx) [reply dd-java-agent/instrumentation/vertx/vertx.gradle]: Good catch! Fixed.

**Pull request merged**